### PR TITLE
refactor(macros): Rename {{DefaultAPISidebar}} to {{APISpecSidebar}}

### PR DESCRIPTION
**Pull request by ExE-Boss**
This PR renames `{{DefaultAPISidebar}}` to `{{APISpecSidebar}}`, as was discussed [in this Discourse thread](https://discourse.mozilla.org/t/40210).

---

I originally intended to include a shim wrapper named `DefaultAPISidebar.ejs`, which would’ve been a simple `template(…)` call (optionally with a `blockIndicator warning` box):
```ejs
<%-await template('APISpecSidebar', arguments)%>
```
But squashing that commit into this one would have broken `git blame`, as git has no real concept of copied or renamed files, unlike mercurial.

I ultimately decided to postpone that into a separate PR.

---

review?(@wbamberg)

### CODE REVIEW

File: tests/macros/APISpecSidebar.test.js [hunk 1]
```diff
@@ -271,44 +321,43 @@ function checkResult(html, config) {
...
+        macro.ctx.template = jest.fn(async (name, ...args) => {
+            if (name === 'L10n:Common') {
+                return commonL10nFixture;
             }
-            if (name === "GroupData") {
+            if (name === 'GroupData') {
                 return groupDataFixture;
             }
-            return await originalTemplate(name, ...args);
+            return originalTemplate(name, ...args);
```
Comment (ExE-Boss) [inline tests/macros/APISpecSidebar.test.js]: `return await` is unnecessary and can lead to degraded performance.

## See also:
- https://eslint.org/docs/rules/no-return-await

File: tests/macros/APISpecSidebar.test.js [hunk 2]
```diff
@@ -271,44 +321,43 @@ function checkResult(html, config) {
...
         // env.slug only has to be truthy
         macro.ctx.env.slug = 'not undefined';
         // Mock calls to L10n-Common and GroupData
         const originalTemplate = macro.ctx.template;
-        macro.ctx.template = jest.fn( async (name, ...args) => {
-            if (name === "L10n:Common") {
-                return commonl10nFixture;
+        macro.ctx.template = jest.fn(async (name, ...args) => {
+            if (name === 'L10n:Common') {
+                return commonL10nFixture;
```
Comment (ExE-Boss) [inline tests/macros/APISpecSidebar.test.js]: This wasn’t correctly formatted.

**Comment by peterbe** [comment]
@ExE-Boss Why do you think it would "break `git blame`"?

**Comment by peterbe** [comment]
I appreciate that once merged, the commit messages listed in git blame won't match what you called it but all the lines will still be there.

**Comment by ExE-Boss** [comment]
That’s because I&nbsp;originally intended to&nbsp;add a&nbsp;shim wrapper for&nbsp;`{{DefaultAPISidebar}}`, but&nbsp;then&nbsp;I&nbsp;ultimately decided against&nbsp;it.

The file would have just been:
```ejs
<%-await template('APISpecSidebar', arguments)%>
```

**Comment by wbamberg** [comment]
This:

Screen Shot 2019-05-17 at 8 51 18 AM

... is not a polite way to interact with other people working on this project. It reads aggressively, as if you are dictating the way maintainers must handle your PR. If you have particular requirements about how you'd like a PR to be merged please discuss it politely, including a clear explanation for why you think it's important in this particular case.

**Comment by wbamberg** [comment]
This PR contains changes that aren't specific to the problem being addressed. Please don't do this.

Here's why. All the people who maintain this repo have a lot to do and reviewing PRs is a significant piece of their time. Ideally, before a PR is merged there's some agreement that the change it makes is desirable - like a bug or forum thread that seems to have reached a consensus. Then if the PR does exactly what was agreed, the review is easy: the maintainer already knows that this change is desirable, and just has to check that the PR contains the right changes, and merge it.

If the PR contains that change mixed up with other unrelated changes, the reviewer now has to figure out:
(1) which bits of the PR make the previously agreed change
(2) what all the other bits do
(3) whether we actually want to do these other bits

The PR has gone from a 5 minute "yes the changes are good" to "what does this all do?" "do we want to have a convention for this?" "might this change regress something else?"

Some of the changes for this PR include type comments:

```
* @callback ItemChecker
* @param {string} expected
* @param {Node & ParentNode} actual
* @param {Locale} locale
* @return {void}
```

In this case, adding type information is a change of policy about coding conventions. Changing conventions requires discussion among project participants who will have to abide by them. 

If you think we should use these comments in KS, start a conversation about it. File an issue in the rep or start a Discourse thread, describing what you want to change and why. If there is consensus, we can file PRs that make exactly those changes. 

One of your previous PRs added type comments as an unrelated change, and David left a comment questioning these and asking for feedback from me and Ryan, who both expressed the view that we didn't want to add these: https://github.com/mdn/kumascript/pull/1093#pullrequestreview-210940603, https://github.com/mdn/kumascript/pull/1093#issuecomment-469899913, https://github.com/mdn/kumascript/pull/1093#issuecomment-469924804. Given that history I do not think it is OK to be reintroducing these changes without any further discussion of it. It feels as if you are trying to force your own views on the maintainers of this repo.

**Comment by ExE-Boss** [comment]
@wbamberg
> ... is not a polite way to interact with other people working on this project. It reads aggressively, as if you are dictating the way maintainers must handle your PR. If you have particular requirements about how you'd like a PR to be merged please discuss it politely, including a clear explanation for why you think it's important in this particular case.

Sorry about that, I originally intended to include a shim wrapper named `DefaultAPISidebar.ejs`, which would’ve been a simple `template(…)` call:
```ejs
<%-await template('APISpecSidebar', arguments)%>
```
But squashing that commit into this one would have broken `git blame`, as git has no real concept of copied or renamed files, unlike mercurial.

**Pull request closed without merging**